pandas/_libs/internals.pyx

```diff
@@ -899,5 +899,8 @@ cdef class BlockValuesRefs:
         -------
         bool
         """
+        self.referenced_blocks = [
+            ref for ref in self.referenced_blocks if ref() is not None
```

Comment:
Maybe we can delete expired weakrefs too?

Reply:
? this checks if a wekref is dead, ref() evaluates to None if the reference is dead

Reply:
Might misunderstand you, but this does exactly this

Reply:
Nvm, misunderstood.